pystac/catalog.py

```diff
@@ -344,13 +345,33 @@ def get_child(
             recursive : If True, search this catalog and all children for the
                 item; otherwise, only search the children of this catalog. Defaults
                 to False.
+            sort_links_by_id : If True, links containing the ID will be checked
+                first. If links doe not contin the ID then setting this to False
+                will improve performance. Defaults to True.
 
         Return:
             Optional Catalog or Collection: The child with the given ID,
             or None if not found.
         """
         if not recursive:
-            return next((c for c in self.get_children() if c.id == id), None)
+            children: Iterable[Union["Catalog", pystac.Collection]]
+            if not sort_links_by_id:
+                children = self.get_children()
+            else:
+
+                def sort_function(id: str, links: List[Link]) -> List[Link]:
+                    return sorted(
+                        links,
+                        key=lambda x: (href := x.get_href()) is None or id not in href,
+                    )
+
+                children = map(
+                    lambda x: cast(Union[pystac.Catalog, pystac.Collection], x),
+                    self.get_stac_objects(
+                        pystac.RelType.CHILD, modify_links=partial(sort_function, id)
```

Comment:
```suggestion
                def sort_function(links: List[Link]) -> List[Link]:
                    return sorted(
                        links,
                        key=lambda x: (href := x.get_href()) is None or id not in href,
                    )

                children = map(
                    lambda x: cast(Union[pystac.Catalog, pystac.Collection], x),
                    self.get_stac_objects(
                        pystac.RelType.CHILD, modify_links=sort_function
```

Can we use variable capture here to remove the `partial`? I know you can in other languages but haven't tested this in Python.

Reply:
Oh good point. Yeah that should work fine.